### PR TITLE
chore: add tsdoc to sanity package

### PR DESCRIPTION
**Pull request by RitaDias**
### Description

WIP

### What to review

- [x] Have tsdoc installed
- [x] Have `@hidden` tag
- ↪️  Automate way to have permissions to run publish for tsdoc #4584 (non blocker)
- ↪️  Write `@public` docs (out of scope for this PR)

### CODE REVIEW

File: packages/sanity/src/core/config/document/actions.ts [hunk 1]
```diff
@@ -49,47 +57,55 @@ export interface DocumentActionConfirmDialogProps {
...
+/**
+ * @hidden
+ * @beta */
 export interface DocumentActionPopoverDialogProps {
   type: 'popover'
   content: React.ReactNode
   onClose: () => void
 }
 
-/** @beta */
```
Comment (binoy14) [inline packages/sanity/src/core/config/document/actions.ts:79]: seems like this got wiped during a rebase I am guessing causing build failure

File: packages/sanity/tsdoc.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        "@hidden": true,
+        "@beta": true,
+        "@internal": true,
+        "@public": true,
+        "@experimental": true,
+        "@see": true,
+        "@link": true,
+        "@example": true,
+        "@deprecated": true,
+        "@alpha": true
```
Comment (binoy14) [inline packages/sanity/tsdoc.json:21]: I believe this is missing `@param` and `@returns` tag causing eslint error

```suggestion
        "@alpha": true,
        "@param": true,
        "@return": true
```

File: packages/sanity/src/core/config/document/actions.ts (hunk 1 above)
Comment (RitaDias) [reply packages/sanity/src/core/config/document/actions.ts:79]: Eagle eyes, thank you 🙏

Comment (RitaDias) [reply packages/sanity/src/core/config/document/actions.ts:79]: Should be re-added now, once more thanks 🙏

File: packages/sanity/tsdoc.json (hunk 2 above)
Comment (binoy14) [reply packages/sanity/tsdoc.json]: made a typo in code suggestion 🤦 pushed a fix

**Comment by binoy14** [comment]
This should be good to review and eventually merge, thoughts @RitaDias ?

If this is merged writing more docs would be easier to test directly against `next` and can be incremental instead of making this PR very large to review both by the team and by Martin

**Comment by RitaDias** [comment]
> This should be good to review and eventually merge, thoughts @RitaDias ?
> 
> If this is merged writing more docs would be easier to test directly against `next` and can be incremental instead of making this PR very large to review both by the team and by Martin

I'm all for it, I think it has little risk. there will be some things that we probably want to change in the test-studio but again it's the test studio so I'm not concerned.

@bjoerge would you be ok with that?

File: dev/test-studio/package.json [hunk 3]
```diff
@@ -24,6 +24,7 @@
     "@sanity/image-url": "^1.0.2",
     "@sanity/portable-text-editor": "3.12.1",
     "@sanity/types": "3.12.1",
+    "@sanity/tsdoc": "1.0.0-alpha.22",
```
Comment (binoy14) [inline dev/test-studio/package.json:27]: Probably should upgrade to latest alpha everywhere before merging

**Review by binoy14 — APPROVED** [review]
✅

**Review by bjoerge — APPROVED** [review]
Great work! 👏🏼

**Pull request merged**